gridded_etl_tools/utils/zarr_methods.py

```diff
@@ -571,32 +571,56 @@ def to_zarr(self, dataset: xr.Dataset, *args, **kwargs):
-                # (there's probably a better way to dothis? compute=False or zarr.consolidate_metadata?).
-                dataset.attrs["update_in_progress"] = True
-                empty_dataset = dataset
-                for coord in chain(dataset.coords, dataset.data_vars):
-                    empty_dataset = empty_dataset.drop(coord)
-                # If there is an existing Zarr, indicate in the metadata that an update is in progress, and write the
-                # metadata before starting the real write.
-                # Note that update_is_append_only is also written here because it was set outside of to_zarr.
-                if self.store.has_existing:
-                    self.info("Pre-writing metadata to indicate an update is in progress")
-                    empty_dataset.to_zarr(self.store.mapper(refresh=True), append_dim=self.time_dim)
+            # Don't use update-in-progress metadata flag on IPLD or on a dataset that doesn't have existing data stored
+            exists_at_start = self.store.has_existing
+            if not isinstance(self.store, IPLD) and exists_at_start:
+                # Update metadata on disk with new values for update_in_progress and update_is_append_only, so that if
+                # a Zarr is opened during writing, there will be indicators that show the data is being edited.
+                self.info("Writing metadata before writing data to indicate write is in progress.")
+                self.store.write_metadata_only(
+                    update_attrs={
+                        "update_in_progress": True,
+                        "update_is_append_only": dataset.get("update_is_append_only"),
+                    }
+                )
+                # Remove update attributes from the dataset putting them in a dictionary to be written post-parse
+                post_parse_attrs = self.post_parse_attrs(dataset=dataset)
 
             # Write data to Zarr and log duration.
             start_writing = time.perf_counter()
             dataset.to_zarr(*args, **kwargs)
             self.info(f"Writing Zarr took {datetime.timedelta(seconds=time.perf_counter() - start_writing)}")
 
             # Don't use update-in-progress metadata flag on IPLD
-            if not isinstance(self.store, IPLD):
+            if not isinstance(self.store, IPLD) and exists_at_start:
                 # Indicate in metadata that update is complete.
-                empty_dataset.attrs["update_in_progress"] = False
-                self.info("Re-writing Zarr to indicate in the metadata that update is no longer in process.")
-                empty_dataset.to_zarr(self.store.mapper(), append_dim=self.time_dim)
+                self.info("Writing metadata after writing data to indicate write is finished.")
+                self.store.write_metadata_only(update_attrs=post_parse_attrs)
+
+    def post_parse_attrs(self, dataset: xr.Dataset = None) -> dict[str, Any]:
```

Comment:
1. I think this should be rewritten to be more functional instead of relying on a [side effect](https://en.wikipedia.org/wiki/Side_effect_(computer_science)). In other words, it should return the dataset with the attrs removed as opposed to just removing them. Then the function name could be something like `move_post_parse_attrs_to_dict`.
2. I don't think `dataset` should be an optional key word arg, I think it's required.